File: lab2.cpp
//digital roots sequence

#include <iostream>

using namespace std;

int main() {
    long long num;
    cout << "enter a number greater than 9 ";
    cin >> num;

    if (num <= 9) {
        cout << "enter a number greater than9" << endl;
        return 1;
    }

    int steps = 0;
    cout << num;

    while (num > 9) {
        long long digit_sum = 0;
        long long temp = num;


        while (temp > 0) {
            digit_sum += temp % 10;
            temp /= 10;
        }

        num = digit_sum;
        steps++;
        cout << " -> " << num;
    }

    cout << "\nfinalvalue " << num << endl;
    cout << "steps taken: " << steps << endl;

    return 0;
}

File: Lab2/lab2.py
#digitalrootsequance
num = int(input("enter a number greater than 9 "))

if num <= 9:
    print("enter a number greater than 9")
else:
    steps = 0
    print(num, end="")

    while num > 9:
        digit_sum = 0
        temp = num


        while temp > 0:
            digit_sum += temp % 10
            temp //= 10

        num = digit_sum
        steps += 1

        print(" →", num, end="")

    print(" final value: ", num)
    print("steps counter: ", steps)

